staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -577,7 +577,16 @@ func (s *SecureServingInfo) HostPort() (string, int, error) {
 // AuthorizeClientBearerToken wraps the authenticator and authorizer in loopback authentication logic
 // if the loopback client config is specified AND it has a bearer token.
 func AuthorizeClientBearerToken(loopback *restclient.Config, authn *AuthenticationInfo, authz *AuthorizationInfo) {
-	if loopback == nil || authn == nil || authz == nil || authn.Authenticator == nil && authz.Authorizer == nil || len(loopback.BearerToken) == 0 {
+	if loopback == nil || len(loopback.BearerToken) == 0 {
+		return
+	}
+	if authn == nil || authz == nil {
+		// prevent nil pointer panic
```

Comment:
We are returning here now if either of the two is nil. I.e. we disable bearer token authz if authn is not plumbed the normal way through. I don't think this is an too important code path. But for complete and correctness we might want to skip line 593 or 596 respectively.

/cc @liggitt

Reply:
previously we required both to be nil? what is the behavior with only authn or only authz configured?

Reply:
> what is the behavior with only authn or only authz configured?

- __authn enabled && authz disabled__: this is useful for AA servers. w/ authz disabled, the server will disable delegated (subject review) authorizer (for our case, we place the request validation into one compiled-in admission controller since it's less costy). and w/ authn enabled, the user's info will still be passed as admission attributes so that the admission could work.
- __authn disabled && authz enabled__: it doesnt seem to work. iiuc, authn should be a prerequisites to authz?